test/tracerEq/test_consistency_2d.py

```diff
@@ -85,42 +90,55 @@ def run_tracer_consistency(constant_c = True, **model_options):
     if options.solve_tracer:
         tracer_int, tracer_int_rerr = solver_obj.callbacks['export']['tracer_2d mass']()
         assert abs(tracer_int_rerr) < 1e-4, 'tracer is not conserved'
+        tracer_mass_int, tracer_mass_int_rerr = solver_obj.callbacks['timestep']['tracer_2d total mass']()    
+        print_output("Tracer total mass error: %11.4e" %(tracer_mass_int_rerr))
+        assert abs(tracer_mass_int_rerr) < 1.2e-4, 'total tracer mass is not conserved'
         smin, smax, undershoot, overshoot = solver_obj.callbacks['export']['tracer_2d overshoot']()
         max_abs_overshoot = max(abs(undershoot), abs(overshoot))
         overshoot_tol = 1e-12
-        msg = 'Tracer overshoots are too large: {:}'.format(max_abs_overshoot)
-        assert max_abs_overshoot < overshoot_tol, msg
-
+        if options.use_tracer_conservative_form == False:
+            msg = 'Tracer overshoots are too large: {:}'.format(max_abs_overshoot)
+            assert max_abs_overshoot < overshoot_tol, msg
 
 
 def test_const_tracer():
     """
     Test CrankNicolson timeintegrator without slope limiters
     Constant tracer, should remain constant
     """
-    run_tracer_consistency(constant_c= True,
+    run_tracer_consistency(constant_c=True,
                            use_nonlinear_equations=True,
                            solve_tracer=True,
                            use_limiter_for_tracers=False,
                            no_exports=True)
 
 
-
 def test_nonconst_tracer():
     """
     Test CrankNicolson timeintegrator  with slope limiters
     Non-trivial tracer, should see no overshoots and be conserved
     """
-    run_tracer_consistency(constant_c= False,
+    run_tracer_consistency(constant_c=False,
                            use_nonlinear_equations=True,
                            solve_tracer=True,
                            use_limiter_for_tracers=True,
-                           no_exports=True)
+                           no_exports=True)   
 
+def test_nonconst_tracer_conservative():
+    """
+    Test CrankNicolson timeintegrator with no slope limiters
+    Non-trivial tracer, should see no overshoots and be conserved
+    """
+    run_tracer_consistency(constant_c=False,
+                           use_nonlinear_equations=True,
+                           solve_tracer=True,
+                           use_limiter_for_tracers=False,
+                           no_exports=True,
+                           use_tracer_conservative_form=True)
 
 if __name__ == '__main__':
-    run_tracer_consistency(constant_c= False,
+    run_tracer_consistency(constant_c=False,
                            use_nonlinear_equations=True,
                            solve_tracer=True,
                            use_limiter_for_tracers=False,
-                           no_exports=False)
+                           no_exports=True)
```

Comment:
This bit is not part of the test suite - only if you run it by hand as a stand-alone "script". So here it is actually useful to have exports switched on.

Reply:
Changed back to how it was. Thanks for the comment!